tests/upgrades/scenario_workers.py

```diff
@@ -0,0 +1,63 @@
+import json
+from pathlib import Path
+
+import pytest
+from automation_tools.satellite6.hammer import set_hammer_config
+from fabric.api import env
+
+import robottelo
+from robottelo.config import settings
+
+
+_json_file = 'upgrade_workers.json'
+
+
+def save_worker_hostname(test_name):
+    data = {}
+    json_file = Path(_json_file)
+    if json_file.exists():
+        data = json.loads(json_file.read_text())
+    # Removing the parameter name from test name before save
+    test_name = test_name.split('[')[0] if '[' in test_name else test_name
+    data.update({test_name: settings.server.hostname})
+    json_file.write_text(json.dumps(data))
+
+
+def get_worker_hostname(test_name):
+    with open(_json_file) as rjson:
+        data = json.load(rjson)
```

Comment:
Consider opening an reading this file at the module level, or in a session scoped fixture. (if it exists, safely)

That way you don't open and JSON parse the file once for every test function in the post_upgrade session.